rules/engine.go

```diff
@@ -118,6 +121,10 @@ func (e *v3Engine) SetKVWrapper(kvWrapper WrapKV) {
 	e.kvWrapper = kvWrapper
 }
 
+func (e *v3Engine) SetWatchWrapper(kvWatcher WrapWatcher) {
+	e.kvWatcher = kvWatcher
```

Comment:
agreeing with out offline conversation, I don't think this is being propagated down to when the watchers are instantiated.  I believe that needs to be done here: https://github.com/IBM-Cloud/go-etcd-rules/blob/master/rules/watcher.go#L16

Once that's done, we should add an integration test to verify the wrapper can observe events as expected and no resources are leaked.